vocabulary-draft/continuous-delivery.md

```diff
@@ -0,0 +1,114 @@
+
+- **TaskRun Started**: a TaskRun inside a PipelineRun has started.
+- **TaskRun Finished**: a TaskRun inside a PipelineRun has finished. 
+
+Pipeline Events MUST include the following attributes:
+- **Event Type**: the type is restricted to include `CD.Pipeline**` prefix. For example `CD.Pipeline.PipelineRun.Queued` or `CD.Pipeline.Tests.Started`
+- **PipelineRun Id**: unique identifier for a pipeline execution
+- **Pipeline Name**: unique identifier for the pipeline, not for the instance. A pipeline can have multiple instances/runs.  
+- **PipelineRun Status**: current status of the PipelineRun at the time when the event was emitted. If the pipeline is finished, this attribute should reflect if it finished successfully or if there was an error on the execution. Possible statuses: [Running, Finished, Error]
+
+Optional attributes: 
+- **PipelineRun URL**: URL to locate where  pipeline instances are running
+- **PipelineRun Errors**: error field to indicate possible compilation, test, build and package errors.
+
+## Product Composition Events
+
+When the products delivered to users can be, or needs to be, broken down into components (hardware, software, configuration files etc.), the creation of versions of said components and the compositoning of components into products or product subsystems may warrant the use of separate events detailed in this section.
+
+Product Composition Events MUST include the following attributes:
+
+- **Event Type**: the type is restricted to include `CD.ProductComposition` prefix. For example `CD.ProductComposition.ComponentVersionCreated` or `CD.ProductComposition.CompositionCreated`
+
+### Component Version Created
+
+Announces that a new version of a Product Component has been created.
+
+Examples of Component Versions include:
+
+- New version of a software module or application.
+- New version of a document or piece of configuration.
+- New revision of a hardware component.
+
+The purpose of this event would be for systems to start creating Product Compositions / Baseline Versions that include this Component Version.
+
+Component Version Created Events MUST include the following attributes:
+
+- **Component Version ID**: Unique identifier of component version.
+- **Component ID**: Identifier of component of which this is a version.
+
+Optional attributes:
+
+- **Version Name**: Machine-friendly and human-friendly version tag/number/etc. for component version, e.g. “1.4.2-alpha2”.
+- **Component Name**: Machine-friendly and human-friendly name of component, e.g. “c37-backbone-router”.
+- **Component Display Name**: UX/human-friendly display name of component, e.g. “C37 Backbone Router”.
+- **Artifacts**: List of artifacts (files, packages, Docker images etc.) part of this component version.
```

Comment:
@erkist so here I can see the relationship, a Component is just a collection of "artefacts" that also has its own version? Can we just not have parent "artefacts" for that?

Reply:
@salaboy A component can optionally be just a collection of artifacts, yes. But this is not mandatory, so I don't agree that it can be replaced by parent/child artifacts.